src/extension.ts

```diff
@@ -65,13 +70,20 @@ function activateTOC(
   // Add the ToC widget to the application restorer:
   restorer.add(toc, 'juputerlab-toc');
 
-  // Create a notebook generator:
-  const notebookGenerator = createNotebookGenerator(
-    notebookTracker,
-    toc,
-    rendermime.sanitizer
-  );
-  registry.add(notebookGenerator);
+  // Load the configuration of whether to enable collapsing behavior
+  settingRegistry.load('@jupyterlab/toc:plugin').then(settings => {
```

Comment:
The notebook generator should not have to depend on loading settings. Settings can fail to load. For an example, see [here](https://github.com/jupyterlab/jupyterlab-git/blob/e923a5beeb5f29ab16085e4a28efeae785128280/src/index.ts#L74) and [here](https://github.com/jupyterlab/jupyterlab-git/blob/e923a5beeb5f29ab16085e4a28efeae785128280/src/model.ts#L21) for how to accommodate async resolution of settings, if possible.